source/fail.js

```diff
@@ -0,0 +1,35 @@
+import _curry2 from './internal/_curry2';
+import _assertPromise from './internal/_assertPromise';
+
+
+/**
+ * Returns the result of applying the onFailure function to the value inside
+ * a failed promise. This is useful for handling rejected promises
+ * inside function compositions.
+ *
+ * @func
+ * @memberOf R
+ * @category Function
+ * @sig (e -> b) -> (Promise e a) -> (Promise e b)
+ * @sig (e -> (Promise f b)) -> (Promise e a) -> (Promise f b)
+ * @param {Function} onFailure The function to apply. Can return a value or a promise.
```

Comment:
"Can return a value or a promise of a value."

Does this mean we are signing up for `then` as collapsed `map/chain`?

Reply:
Well, `then` _is_ a collapsed `map/chain`. I tried to add the signatures for each case. It doesn't make sense to only support one or the other, I can't think of which one we would prioritize.